tests/unittests/tests-unicoap/tests-unicoap-pdu.c

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (C) 2024-2025 Carl Seifert
+ * Copyright (C) 2024-2025 TU Dresden
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @file
+ * @ingroup unittests
+ * @brief   Unit tests for PDU parsers
+ * @author  Carl Seifert <carl.seifert1@mailbox.tu-dresden.de>
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "tests-unicoap.h"
+
+#include "net/unicoap/message.h"
+#include "net/unicoap/options.h"
+
+#define _TEST_ASSERT_EQUAL_PARSED(parsed, code, payload, token) \
+    test_parsed_message((parsed), (uint8_t)code, sizeof_string(payload), (uint8_t*)payload, sizeof(token), token)
+
+#define _TEST_ASSERT_EQUAL_PARSED_RFC7252(parsed, code, payload, token, message_type, message_id) \
+    _TEST_ASSERT_EQUAL_PARSED((parsed), (code), (payload), (token));                              \
+    TEST_ASSERT_EQUAL_INT((parsed)->properties.rfc7252.id, (message_id));                         \
+    TEST_ASSERT_EQUAL_INT((parsed)->properties.rfc7252.type, (message_type))
+
+static inline void test_parsed_message(unicoap_parser_result_t* parsed, uint8_t code, size_t payload_size, const uint8_t* payload, uint8_t token_length, uint8_t* token)
```

Comment:
```suggestion
static void test_parsed_message(unicoap_parser_result_t* parsed, uint8_t code, size_t payload_size, const uint8_t* payload, uint8_t token_length, uint8_t* token)
```

or why would you want to inline and increase the `.text` segment?